tests/openvino/test_modeling.py

```diff
@@ -578,6 +578,72 @@ def test_compare_to_transformers(self, model_arch):
+            not_stateful.append("mixtral")
+
+        if is_openvino_version("<", "2024.1"):
+            not_stateful.extend(["llama", "gemma"])
+
+        if "gptq" in model_arch:
+            self.skipTest("GPTQ model loading unsupported with AutoModelForCausalLM")
+        if model_arch in ["chatglm", "baichuan2"]:
+            self.skipTest("Models " + model_id + "doesn't support concurrent execution in AutoModelForCausalLM")
+
+        set_seed(SEED)
+        model_kwargs = {}
+        if model_arch in self.REMOTE_CODE_MODELS:
+            model_kwargs = {"trust_remote_code": True}
+
+        ov_model = OVModelForCausalLM.from_pretrained(model_id, export=True, ov_config=F32_CONFIG, **model_kwargs)
+        self.assertIsInstance(ov_model.config, PretrainedConfig)
+        self.assertTrue(ov_model.use_cache)
+        self.assertEqual(
+            ov_model.stateful, self.IS_SUPPORT_STATEFUL and ov_model.config.model_type not in not_stateful
+        )
+
+        transformers_model = AutoModelForCausalLM.from_pretrained(model_id, **model_kwargs)
+        tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=model_arch in self.REMOTE_CODE_MODELS)
+        if model_arch == "qwen":
+            transformers_model.to(torch.float32)
+        inputs_list = ["This is a cat", "This is a dog", "Yet another test"]
+        tokens_list = [
+            tokenizer(inputs, return_tensors="pt", return_token_type_ids=False if model_arch == "llama" else None)
+            for inputs in inputs_list
+        ]
+
+        def run_ov_model(tokens, transformers_model, ov_model):
+            # global ov_model, transformers_model
+            # position_ids = None
+            # if model_arch.replace("_", "-") in MODEL_TYPES_REQUIRING_POSITION_IDS:
+            #    input_shape = tokens["input_ids"].shape
+            #    position_ids = (
+            #        torch.arange(0, input_shape[-1], dtype=torch.long).unsqueeze(0).view(-1, input_shape[-1])
+            #    )
+            set_seed(SEED)
+            ov_outputs = ov_model(**tokens)
+
+            self.assertTrue("logits" in ov_outputs)
+            self.assertIsInstance(ov_outputs.logits, torch.Tensor)
+            # self.assertTrue("past_key_values" in ov_outputs)
+            # self.assertIsInstance(ov_outputs.past_key_values, tuple)
+            # if self.IS_SUPPORT_STATEFUL and model_arch != "gpt_bigcode":
+            #    self.assertTrue(len(ov_outputs.past_key_values) == 1 and len(ov_outputs.past_key_values[0]) == 0)
+            with torch.no_grad():
+                transformers_outputs = transformers_model(**tokens)
+            # Compare tensor outputs
+            self.assertTrue(torch.allclose(ov_outputs.logits, transformers_outputs.logits, atol=1e-4))
+            # self.assertTrue(False)
+
+        run_on_multiple_threads(run_ov_model, tokens_list, (transformers_model, ov_model))
+
+        del transformers_model
+        del ov_model
+        gc.collect()
```

Comment:
The time taken to run all tests is already non negligible so I think we should merge it with `test_compare_to_transformers` (to not duplicate steps like export)